lib/spark/component/attribute.rb

```diff
@@ -70,6 +75,28 @@ def initialize_attributes(attrs = nil)
         end
       end
 
+      def initialize_attribute_default_groups(attrs)
+        self.class.attribute_default_groups.each do |name, options|
+          # Determine what was passed for this argument group
+          choice = attrs[name] || self.class.attributes[name]
+
+          # Get attributes to set from argument
+          option = options[choice]
+
+          next unless option
+          unless option.is_a?(Hash)
+            raise("In argument group `:#{name}`, value `#{option}` must be a hash.")
```

Comment:
We should probably change the error message here to remove "argument group".

Reply:
Good catch!